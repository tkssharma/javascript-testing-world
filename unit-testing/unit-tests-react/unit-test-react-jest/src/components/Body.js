import React, { useState } from "react";
const Body = ({title}) => {
    //local state
    const [count, setCount] = useState(0)
    //click event handler for button
    const handleClick = () => {
        setCount(count+1)
    }
    return (
        <div>
            <h1 data-testid="header">{title}</h1>
            <p data-testid="counter">{count}</p>
            <button type="button" onClick={handleClick} data-testid="button">Tap Me</button>
        </div>
    )
}

export default Body;