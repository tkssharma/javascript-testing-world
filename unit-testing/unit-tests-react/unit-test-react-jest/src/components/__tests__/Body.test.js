import React from 'react';
import Body from '../Body';
import {render} from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe("This suit is to test the Body component", ()=>{
    test('Snapshot of Body', () => {
        const {asFragment} = render(<Body title="hello world"/>);
        expect(asFragment()).toMatchSnapshot()
    });   
    
    test("finding title", ()=>{
        const {getByText} = render(<Body title="Jay Rocks"/>);
        expect(getByText("Jay Rocks")).toBeInTheDocument("Jay Rocks")
    })

    test("finding title with TestId", ()=>{
        const {getByTestId} = render(<Body title="Jay Rocks"/>);
        expect(getByTestId("header")).toHaveTextContent("Jay Roc")
        //exact match
        expect(getByTestId("header")).toHaveTextContent(/^Jay Rocks$/)
      })

      test("check Counter initial value", ()=>{
        const {getByTestId} = render(<Body title="Jay Rocks"/>);
        expect(getByTestId("counter")).toHaveTextContent("0")
      })

      test("check Counter value after a tap", ()=>{
        const {getByTestId} = render(<Body title="I Love Kavinie"/>);
        const button = getByTestId("button");
        userEvent.click(button)
        expect(getByTestId("counter")).toHaveTextContent("1")
      })

      test("check Counter value after tapping twice", ()=>{
        const {getByTestId} = render(<Body title="Jay Rocks"/>);
        const button = getByTestId("button");
        userEvent.click(button)
        userEvent.click(button)
        expect(getByTestId("counter")).toHaveTextContent("2")
      })
})