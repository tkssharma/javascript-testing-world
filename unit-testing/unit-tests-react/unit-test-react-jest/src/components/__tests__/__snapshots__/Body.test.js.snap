// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the Body component Snapshot of Body 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="header"
    >
      hello world
    </h1>
    <p
      data-testid="counter"
    >
      0
    </p>
    <button
      data-testid="button"
      type="button"
    >
      Tap Me
    </button>
  </div>
</DocumentFragment>
`;
